src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -3390,44 +3387,10 @@ internal Buffer(IEnumerable<TElement> source, bool queryInterfaces = true)
                 {
                     items = iterator.ToArray();
                     count = items.Length;
-                }
-                else
-                {
-                    ICollection<TElement> collection = source as ICollection<TElement>;
-                    if (collection != null)
-                    {
-                        count = collection.Count;
-                        if (count > 0)
-                        {
-                            items = new TElement[count];
-                            collection.CopyTo(items, 0);
-                        }
-                    }
+                    return;
                 }
             }
-
-            if (items == null)
-            {
-                using (IEnumerator<TElement> e = source.GetEnumerator())
-                {
-                    if (e.MoveNext())
-                    {
-                        items = new TElement[4];
-                        items[0] = e.Current;
-                        count = 1;
-                        
-                        while (e.MoveNext())
-                        {
-                            if (items.Length == count) Array.Resize(ref items, checked(count * 2));
-                            items[count] = e.Current;
-                            ++count;
-                        }
-                    }
-                }
-            }
-
-            this.items = items;
-            this.count = count;
+            items = EnumerableHelpers.ToArray(source, out count);
```

Comment:
The ```queryInterfaces``` parameter no longer has the same meaning it used to.  It previously dictated whether to query for any base/interfaces at all, or whether to always fall back to enumeration, but now regardless of the value ICollection will always be queried for inside of ToArray.  I think that's ok, because the one call site that sets it to false really only cares about whether we check for Iterator, but the name is a misnomer now (I guess it was previously, too, because Iterator is a base class rather than an interface).  Let's rename it to something like "queryIterator" or something better. :smile: